### PR TITLE
Add vsphere machine deployment template

### PR DESCRIPTION
**Pull request by k4leung4**
**What this PR does / why we need it**:
This is to add a template for create machine deployments

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

@kubernetes/kube-deploy-reviewers

### CODE REVIEW

File: clusterctl/examples/vsphere/machinedeployment.yaml.template [hunk 1]
```diff
@@ -0,0 +1,34 @@
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: MachineSet
```
Comment (mkjelland) [inline clusterctl/examples/vsphere/machinedeployment.yaml.template]: Should this be MachineDeployment?

Comment (k4leung4) [reply clusterctl/examples/vsphere/machinedeployment.yaml.template]: my mistake
thanks for catching that.

File: clusterctl/examples/vsphere/machinedeployment.yaml.template [hunk 2]
```diff
@@ -0,0 +1,34 @@
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: MachineDeployment
+metadata:
+  name: test-machineset-1
```
Comment (mkjelland) [inline clusterctl/examples/vsphere/machinedeployment.yaml.template]: This should also be updated to test-machinedeployment-1 :)

Comment (k4leung4) [reply clusterctl/examples/vsphere/machinedeployment.yaml.template]: doh
fixed.

**Comment by karan** [comment]
Please add documentation.

**Comment by roberthbailey** [comment]
ping @k4leung4 - this should be quick & easy to wrap up.

File: clusterctl/examples/vsphere/machinedeployment.yaml.template [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            datacenter: ""
+            datastore: ""
+            resource_pool: ""
+            network: ""
+            num_cpus: "2"
+            memory: "2048"
+            vm_template: ""
+            disk_label: ""
+            disk_size: ""
+            virtual_machine_domain: ""
```
Comment (karan) [inline clusterctl/examples/vsphere/machinedeployment.yaml.template]: This is not needed

**Pull request closed without merging**